weldx/tests/asdf_tests/test_asdf_groove.py

```diff
@@ -1,19 +1,29 @@
 """Test all ASDF groove implementations."""
 import pytest
 from decorator import contextmanager
-from matplotlib import pylab
 
 from weldx.asdf.util import write_read_buffer
-from weldx.constants import Q_
-from weldx.constants import WELDX_UNIT_REGISTRY as UREG
+from weldx.constants import _DEFAULT_LEN_UNIT, Q_
 from weldx.geometry import Profile
 from weldx.welding.groove.iso_9692_1 import (
     IsoBaseGroove,
     _create_test_grooves,
     get_groove,
 )
 
-_DEFAULT_LEN_UNIT = UREG.millimeters
+
+@contextmanager
+def _close_plot():
+    try:
+        from matplotlib import pylab
+    except ImportError:
+        yield
+    else:
+        try:
+            yield
+        finally:
+            pylab.close()
```

Comment:
can you give me a heads up on what this does? @marscher

Reply:
the test runs did not close their figures, leading to a warning of matplotlib. This context manager closes the produced figures by every test run.